pages/instructorScrapPaper/instructorScrapPaper.js

```diff
@@ -0,0 +1,53 @@
+const ERR = require('async-stacktrace');
+const express = require('express');
+const router = express.Router();
+const { createBarcodedPdf } = require('../../lib/scrapPaperMaker');
+const error = require('../../prairielib/lib/error');
+
+const pageLimit = 1000;
+const charLimit = 45;
+
+router.get('/', (req, res) => {
+  res.locals['pageLimit'] = pageLimit;
+  res.locals['charLimit'] = charLimit;
+  res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+});
+
+router.post('/', function (req, res, next) {
+  if (!res.locals.authz_data.has_course_instance_permission_edit) return next();
```

Comment:
You should include this check for the `get()` router too. Also, you should specifically `403` here with a meaningful error message. See https://github.com/PrairieLearn/PrairieLearn/blob/c4a120a217800c8dc4978b993cbc3ab2f1bffb10/pages/courseSyncs/courseSyncs.js#L20 for an example.

Bonus points if you convert this router to use `async`/`await` 🙂

Reply:
Done. But not the async / await... unless the routes have been setup to work with promises.